### PR TITLE
Updated the README.md to contain more info about the package

### PR DESCRIPTION
**Pull request by Swey9**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,36 @@
...
 
-## Normal Usage
-The normal usage is the same as in argcomplete, only changing the import from argcomplete to cachedcomplete
+## Installation
+
+Use the package manager pip to install cachedcomplete.
+
+```bash
+pip install cachedcomplete
+```
```
Comment (Roynecro97) [inline README.md]: @Swey9 Should we also mention the `activate-global...` part?

File: README.md [hunk 2]
```diff
@@ -1,11 +1,36 @@
...
+
+## Explenation
```
Comment (Roynecro97) [inline README.md]: *Explanation
(might want to install a spell checker)

File: README.md [hunk 3]
```diff
@@ -63,4 +98,28 @@ class CustomCompletionFinder(argcomplete.CompletionFinder):
...
+add a comment with the wanted files and the prefix CACHEDCOMPLETE_HASH:
+
+```python
+# CACHEDCOMPLETE_HASH: file1.py
+```
+
+It also allows multiple files split to multiple comments, or within one comment.
+
+```python
+# CACHEDCOMPLETE_HASH: file1.py, file2.py
```
Comment (Roynecro97) [inline README.md]: No commas, separate with spaces - shell-like syntax

And maybe rename a file to be `.json` (or something else) instead to emphasize not only python files are supported?

File: README.md [hunk 4]
```diff
@@ -63,4 +98,28 @@ class CustomCompletionFinder(argcomplete.CompletionFinder):
...
+# CACHEDCOMPLETE_HASH: file3.py
+# CACHEDCOMPLETE_HASH: "file with spaces.py"
+```
+
+And is also allows passing up a directory to track all the files within
```
Comment (Roynecro97) [inline README.md]: "is also allows" -> "also allows"

File: README.md [hunk 5]
```diff
@@ -1,11 +1,37 @@
+
 # cachedcomplete
+
 Fast semi-static autocomplete for python
```
Comment (nivbromberg) [inline README.md]: Maybe define what is "semi-static"?

File: README.md [hunk 6]
```diff
@@ -1,11 +1,37 @@
...
 
-## Normal Usage
-The normal usage is the same as in argcomplete, only changing the import from argcomplete to cachedcomplete
+## Installation
+
+Use the package manager pip to install cachedcomplete.
```
Comment (nivbromberg) [inline README.md:8]: Consider adding instructions for explicitly running "`setup.py`".
It's common practice with many packages

File: README.md [hunk 7]
```diff
@@ -1,11 +1,37 @@
...
+activate-global-python-argcomplete
+```
+
+## Explanation
+
+cachedcomplete wraps argcomplete, and allows caching of its parsers
+in order to save up time.
+
+Cachedcomplete re-caches everytime a change was made in itself,
+or in the files specified by the user.
```
Comment (nivbromberg) [inline README.md:21]: Maybe explain how caching is done?
Or how changes are tracked/compared?

File: README.md [hunk 8]
```diff
@@ -63,4 +99,28 @@ class CustomCompletionFinder(argcomplete.CompletionFinder):
...
+### Specifying what files to track
+
+In order to specify which files to track changes in,
+add a comment with the wanted files and the prefix CACHEDCOMPLETE_HASH:
+
+```python
+# CACHEDCOMPLETE_HASH: file1.py
+```
+
+It also allows multiple files split to multiple comments, or within one comment.
```
Comment (nivbromberg) [inline README.md:113]: _Nice_

File: README.md [hunk 9]
```diff
@@ -63,4 +99,28 @@ class CustomCompletionFinder(argcomplete.CompletionFinder):
...
+
+```python
+# CACHEDCOMPLETE_HASH: file1.py file2.json
+# CACHEDCOMPLETE_HASH: file3.py
+# CACHEDCOMPLETE_HASH: "file with spaces.txt"
+```
+
+And also allows passing up a directory to track all the files within
```
Comment (nivbromberg) [inline README.md:121]: Maybe add support for glob/regex?

**Review by nivbromberg — COMMENTED** [review]
Looks really nice and explanatory!
I do have some questions/suggestion

File: README.md (hunk 5 above)
Comment (nivbromberg) [reply README.md]: Or maybe I don't know enough about completers :(

File: README.md (hunk 2 above)
Comment (nivbromberg) [reply README.md]: > *Explanation
> (might want to install a spell checker)

I personally recommend "Grammarly"
It's really cool and very helpful and integrates with almost everything.

Comment (nivbromberg) [reply README.md]: > *Explanation
> (might want to install a spell checker)

Test

File: README.md (hunk 5 above)
Comment (nivbromberg) [reply README.md]: > Maybe define what is "semi-static"?

Or maybe I don't know enough about completers 😞

File: README.md (hunk 2 above)
Comment (Roynecro97) [reply README.md]: > I personally recommend "Grammarly"
> It's really cool and very helpful and integrates with almost everything.

They're not so good when using GitHub or editing code.
I meant he should install a VSCode plugin.

File: README.md (hunk 5 above)
Comment (Roynecro97) [reply README.md]: > > Maybe define what is "semi-static"?
> 
> Or maybe I don't know enough about completers 😞

This line should be updated to match the new description.
But I don't think the top of the README is the place for definitions :/

File: README.md (hunk 9 above)
Comment (Roynecro97) [reply README.md:121]: No. It transforms the code from a simple, short line to a very complicated piece of code.

Comment (nivbromberg) [reply README.md:121]: OK. Your call.
If you want I can try to make it as least ugly as I can.

Comment (Roynecro97) [reply README.md:121]: @nivbromberg You're welcome to try, if it turns out well we'll merge it 😄

**Review by Roynecro97 — APPROVED** [review]

**Pull request merged**